pkg/plugin/interface.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"context"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/hashicorp/go-plugin"
+	"google.golang.org/grpc"
+)
+
+const (
+	handShakeCookieKey   = "JVS_PLUGIN"
+	handShakeCookieValue = "hello"
+	JiraGRPCPlugin       = "jira_grpc_plugin"
+)
+
+// Handshake is a common handshake that is shared by plugin and host.
+// handshakeConfigs are used to just do a basic handshake between
+// a plugin and host. If the handshake fails, a user friendly error is shown.
+// This prevents users from executing bad plugins or executing a plugin
+// directory. It is a UX feature, not a security feature.
+var HandShake = plugin.HandshakeConfig{
+	// This isn't required when using VersionedPlugins
+	ProtocolVersion:  1,
+	MagicCookieKey:   handShakeCookieKey,
+	MagicCookieValue: handShakeCookieValue,
+}
+
+// PluginMap is the map of plugins we can dispense.
+var PluginMap = map[string]plugin.Plugin{
+	JiraGRPCPlugin: &ValidatorGRPCPlugin{},
```

Comment:
This doesn't look like need to be unique across plugins. Shall we remove "jira" from it?

Reply:
It doesn't look like so. Per Seth's comment https://github.com/abcxyz/jvs/pull/288#discussion_r1227239343, each plugin will have its own `plugin.Plugin` and this map will be the same across plugins.

Reply:
Did that in https://github.com/abcxyz/jvs/pull/292